### PR TITLE
Add Cosmos support to Utils and SDK

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Adds Cosmos support to the SDK
- Implements CosmWasm and Cosmos adapters for use in the Nexus UI

### Backward compatibility

Yes

### Testing

- Unit Tests for chainmetadata and provider changes
- Manual for contract interactions

### CODE REVIEW

File: typescript/sdk/src/core/adapters/CwCoreAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import { BaseCwAdapter } from '../../app/MultiProtocolApp';
+import { ChainName } from '../../types';
+
+import { ProviderType, TypedTransactionReceipt } from '../../providers/ProviderType';
+import { ICoreAdapter } from './types';
+
+// This adapter just routes to the HyperlaneCore
+// Which implements the needed functionality for Cw chains
+// TODO deprecate HyperlaneCore and replace all Cw-specific classes with adapters
```
Comment (jmrossy) [inline typescript/sdk/src/core/adapters/CwCoreAdapter.ts]: Copy paste cruft, you can delete this comment

File: typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts [hunk 2]
```diff
@@ -38,6 +42,60 @@ import {
 } from './ITokenAdapter';
 
 // Interacts with IBC denom tokens
+export class NativeTokenAdapter
```
Comment (jmrossy) [inline typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts:45]: Suggest moving to a different file for clarity

File: typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts [hunk 3]
```diff
@@ -38,6 +42,60 @@ import {
...
+
+  populateApproveTx(_transferParams: TransferParams): unknown {
+    throw new Error('Approve not required for native tokens');
+  }
+
+  async populateTransferTx(
+    transferParams: TransferParams,
+  ): Promise<MsgTransferEncodeObject> {
+    const transfer: MsgTransfer = {
+      sourcePort: '',
```
Comment (jmrossy) [inline typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts:76]: I'm getting type errors from this line. Does it build in the SDK?

File: typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts [hunk 4]
```diff
@@ -38,6 +42,60 @@ import {
...
+      timeoutHeight: {
+        revisionNumber: 0n,
+        revisionHeight: 0n,
+      },
+      timeoutTimestamp: 0n,
+      memo: '',
+    };
+    return {
+      typeUrl: '/ibc.applications.transfer.v1.MsgTransfer',
+      value: transfer,
```
Comment (jmrossy) [inline typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts:93]: I get a type error here. A mismatch between the `cosmjs-types` MsgTransfer and the @cosmjs one. Do you see that too?

Comment (yorhodes) [reply typescript/sdk/src/token/adapters/CosmWasmTokenAdapter.ts]: yeah ignoring for now

**Review by nambrot — APPROVED** [review]

**Pull request merged**